### DISCUSSION TITLE
Library support for CTAP2_1

### DISCUSSION THREAD

**Discussion by DEX1999** [Yubico/libfido2#742] [Q&A] [answered]
Our company produces devices that support CTAP2_0 and CTAP2_1 but not CTAP2_1PRE. 

Device honestly explicitly tells via authenticatorGetInfo (0x04) command that it supports CTAP2_0 and CTAP2_1 but not CTAP2_1PRE, but when using cred management interface library starts sending 2_1PRE commands (0x41 to be specific). 

After looking through the sources we discovered that libfido2 doesnt support CTAP2_1 at all.

So questions are if support of this protocol version  is planned and when if it is and why does liblrary tries to send unsupported commands to device?

**Comment by LDVG** [comment, marked as answer]
Hi,

Thanks for bringing this to our attention. We certainly also want to support authenticators that do not implement `FIDO_2_1_PRE` functionality and will address this as soon as we are able. I cannot promise any particular time frame at this point though. Patches are welcome.

**Comment by LDVG** [comment]
Hi @DEX1999,

Would you be able to test https://github.com/Yubico/libfido2/pull/751 with your authenticator?

Thanks in advance.

**Reply by DEX1999** [reply]
Yep. Everything works.
